frontend/src/app/(navfooter)/settings/editor/EditorSettings.tsx

```diff
@@ -32,5 +54,27 @@ export default function EditorSettings() {
                 </div>
             </Checkbox>
         </Section>
+        <Section title="Language server">
+            <Checkbox
+                checked={languageServerEnabled}
+                onChange={state => {
+                    setLanguageServerEnabled(state)
+                    setLanguageServerSettingsMutated(true)
+                }}
+                label="Enable language server"
+                description="Enable editor features such as code completion, error checking, and formatting via clangd and WebAssembly magic. WARNING: enabling will incur a one time ~11 - 13MB download, and bump up resource usage during editing.">
+
+                <Checkbox
+                    checked={preferSmallLanguageServer}
+                    onChange={state => {
+                        setPreferSmallLanguageServer(state)
+                        setLanguageServerSettingsMutated(true)
+                    }}
+                    label="Prefer small language server build"
+                    description="Use a smaller (11MB vs 13MB download) build of the language server, at the expense of some performance."/>
+
+                {downloadingLanguageServer && <LoadingSpinner width="24px" />}
```

Comment:
Can you write "Loading" or something next to this? e.g.
```
<div className="pa-4"><LoadingSpinner /> Downloading...</div>
```

Reply:
Done